zone/spells.cpp

```diff
@@ -2044,6 +2044,14 @@ bool Mob::SpellFinished(uint16 spell_id, Mob *spell_target, CastingSlot slot, ui
 	if(!IsValidSpell(spell_id))
 		return false;
 
+	//Cazic Touch targets the pet owner instead of the pet when said pet is tanking.
+	if ((RuleB(Spells, CazicTouchTargetsPetOwner) && spell_target->HasOwner()) && spell_id == DB_SPELL_CAZIC_TOUCH || spell_id == DB_SPELL_TOUCH_OF_VINITRAS) {
```

Comment:
Your `or` will not work as expected here. It will run this logic IF the rule is enabled and `DB_SPELL_CAZIC_TOUCH` OR the spell is `DB_SPELL_TOUCH_OF_VINITRAS`. Basically, it won't matter whether the rule is enabled or not as long as the spell passing through is TOV

Your spell checks with the or's need to be wrapped in parenthesis to encapsulate the OR inside of the AND

**Original**

```cpp
if ((RuleB(Spells, CazicTouchTargetsPetOwner) && spell_target->HasOwner()) && spell_id == DB_SPELL_CAZIC_TOUCH || spell_id == DB_SPELL_TOUCH_OF_VINITRAS) {
```

**Suggested**

```cpp
if ((RuleB(Spells, CazicTouchTargetsPetOwner) && spell_target->HasOwner()) && (spell_id == DB_SPELL_CAZIC_TOUCH || spell_id == DB_SPELL_TOUCH_OF_VINITRAS)) {
```

For context as well OR's and AND's get processed left to right

According to the C++ Standard

```
1 The && operator groups left-to-right. The operands are both contextually converted to bool (Clause 4). The result is true if both operands are true and false otherwise. Unlike &, && guarantees left-to-right evaluation: the second operand is not evaluated if the first operand is false.
```

and

```
1 The || operator groups left-to-right. The operands are both contextually converted to bool (Clause 4). It returns true if either of its operands is true, and false otherwise. Unlike |, || guarantees left-to-right evaluation; moreover, the second operand is not evaluated if the first operand evaluates to true.
```

Reply:
Still need parenthesis wrapped around

```
&& (spell_id == DB_SPELL_CAZIC_TOUCH || spell_id == DB_SPELL_TOUCH_OF_VINITRAS)
```